backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -117,11 +131,14 @@ public boolean registerAssetsInitially() {
         log.info("Registration of item-stock response api successful " + (result = registerApiAsset(DT_ApiMethodEnum.RESPONSE)));
         if (!result) return false;
         log.info("Registration of item-stock status-request api successful " + (result = registerApiAsset(DT_ApiMethodEnum.STATUS_REQUEST)));
+        if (!result) return false;
         if (variablesService.isUseFrameworkPolicy()) {
             log.info("Registration of framework agreement policy successful " + (result = createFrameWorkPolicy()));
         } else {
             log.info("Skipping registration of framework agreement policy");
         }
+        if (!result) return false;
```

Comment:
In case the framework policy is not used (i.e. isUseFrameworkPolicy = false) this line of code does exactly the same thing as line 134, checking the same result a second time. You should consider moving it directly into the if-case.

Reply:
Fixed